contracts/swap/factory.go

```diff
@@ -0,0 +1,123 @@
+package swap
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	contractFactory "github.com/ethersphere/go-sw3/contracts-v0-1-1/simpleswapfactory"
+)
+
+var (
+	// ErrNotDeployedByFactory is given when a contract was not deployed by the factory
+	ErrNotDeployedByFactory = errors.New("not deployed by factory")
+
+	// Deployments maps from network ids to deployed contract factories
+	Deployments = map[uint64]common.Address{
+		// Ropsten
+		3: common.HexToAddress("0x2e9C43E186eaF4fee10799d67e75f8CFc5BA3a0c"),
+	}
+)
+
+type simpleSwapFactory struct {
+	instance *contractFactory.SimpleSwapFactory
+	address  common.Address
+	backend  Backend
+}
+
+// SimpleSwapFactory interface defines the methods available for a factory contract for SimpleSwap
+type SimpleSwapFactory interface {
+	// DeploySimpleSwap deploys a new SimpleSwap contract from the factory and returns the ready to use Contract abstraction
+	DeploySimpleSwap(auth *bind.TransactOpts, issuer common.Address, defaultHardDepositTimeoutDuration *big.Int) (Contract, error)
+	// VerifyContract verifies that the supplied address was deployed by this factory
+	VerifyContract(address common.Address) error
+	// VerifySelf verifies that this is a valid factory on the network
+	VerifySelf() error
+}
+
+// FactoryAt creates a SimpleSwapFactory instance for the given address and backend
+func FactoryAt(address common.Address, backend Backend) (SimpleSwapFactory, error) {
+	simple, err := contractFactory.NewSimpleSwapFactory(address, backend)
+	if err != nil {
+		return nil, err
+	}
+	c := simpleSwapFactory{instance: simple, address: address, backend: backend}
+	return c, err
+}
+
+// FactoryAddressForNetwork gets the default factory address for a given network id
+func FactoryAddressForNetwork(networkID uint64) (common.Address, error) {
+	address, ok := Deployments[networkID]
```

Comment:
We don't know concurrency control here right, because this is only used at startup right?

Reply:
Yes, this only runs once during the constructor of `Swap`.